api/controllers/PushNotificationController.js

```diff
@@ -23,19 +23,37 @@ module.exports = {
     .fetch()
     .then(device => {
       if (device) {
-        return res.ok({result: "Known"});
-      }
-
-      return Device.forge({
-        token: req.param("deviceToken"),
-        user_id: req.session.userId
-      })
-      .save()
-      .then(device => res.ok({result: "Added"}));
+        return device
+        .save({enabled: true})
+        .then(() => res.ok({result: "Known"}))
+      } else {
+        return Device.forge({
+          token: req.param("deviceToken"),
+          user_id: req.session.userId
+        })
+        .save()
+        .then(device => res.ok({result: "Added"}));
+      };
     })
     .catch(res.serverError);
   },
 
+  disableDevice: function (req, res) {
+    var token = req.param('deviceToken');
+    if (!token) {
+      return res.serverError('no device token');
+    }
+
+    return Device.forge({
+      token: token,
+      user_id: req.session.userId
```

Comment:
You can avoid the SELECT entirely here:
`Device.query().where({token: token}).update({enabled: false})`